mergify_engine/web/api/queues.py

```diff
@@ -317,5 +324,435 @@ async def repository_queues(
+
+
+@router.post(
+    "/repos/{owner}/{repository}/queue/{queue_name}/pause_from",  # noqa: FS003
+    summary="Pauses merge queue(s)",
+    description="Pauses the merge of all merge queues equal to and under the requested queue priority level",
+    responses={
+        **api.default_responses,  # type: ignore
+        404: {"description": "Not found"},
+        200: {"description": "The queue(s) have been paused"},
+    },
+)
+async def pause_queues_from(
+    request: requests.Request,
+    application: application_mod.Application = fastapi.Depends(  # noqa: B008
+        security.get_application
+    ),
+    owner: github_types.GitHubLogin = fastapi.Path(  # noqa: B008
+        ..., description="The owner of the repository"
+    ),
+    repository: github_types.GitHubRepositoryName = fastapi.Path(  # noqa: B008
+        ..., description="The name of the repository"
+    ),
+    queue_name: rules.QueueName = fastapi.Path(  # noqa: B008
+        ..., description="The name of the queue"
+    ),
+    redis_cache: utils.RedisCache = fastapi.Depends(  # noqa: B008
+        redis.get_redis_cache
+    ),
+    redis_queue: utils.RedisQueue = fastapi.Depends(  # noqa: B008
+        redis.get_redis_queue
+    ),
+    installation_json: github_types.GitHubInstallation = fastapi.Depends(  # noqa: B008
+        security.get_installation
+    ),
+) -> responses.JSONResponse:
+
+    try:
+        raw_json = await request.json()
+        data = QueuePauseSchema(raw_json)
+        pause_message = data["message"]
+        if pause_message == "":
+            pause_message = "No pause message was specified"
+
+    except voluptuous.Invalid:
+        return responses.JSONResponse(
+            status_code=400, content={"error": "Invalid json"}
+        )
+
+    async with github.aget_client(installation_json) as client:
+        try:
+            # Check this token as access to this repository
+            repo = typing.cast(
+                github_types.GitHubRepository,
+                await client.item(f"/repos/{owner}/{repository}"),
+            )
+        except (http.HTTPNotFound, http.HTTPForbidden, http.HTTPUnauthorized):
+            raise fastapi.HTTPException(status_code=404)
+
+        sub = await subscription.Subscription.get_subscription(
+            redis_cache, repo["owner"]["id"]
+        )
+
+        installation = context.Installation(
+            installation_json, sub, client, redis_cache, redis_queue
+        )
+        repository_ctxt = installation.get_repository_from_github_data(repo)
+        config_file_scheme = await repository_ctxt.get_mergify_config_file()
+
+        if config_file_scheme and installation.redis_queue:
+            config_file = get_mergify_config(config_file_scheme)
+
+            repo_id = repo["id"]
+            owner_id = installation.owner_id
+            queue_rules = config_file["queue_rules"]
+
+            if queue_name not in [rule.name for rule in queue_rules]:
```

Comment:
```suggestion
            if all(queue_name != rule.name for rule in queue_rules):
```
this makes sure you don't need to iterate over all `queue_rules`